SmellDetector/Detectors/RuleEngines/ClassMetricRuleEngine.cs

```diff
@@ -10,34 +11,73 @@ namespace SmellDetector.Detectors.RuleEngines
     internal class ClassMetricRuleEngine : IDetector
     {
         private readonly List<Rule> _rules;
+        private readonly List<Rule> _dynamicRules;
         public ClassMetricRuleEngine()
         {
             Rule rule1 = new Rule("10.1109/WCRE.2005.15",
                                   new AndCriteria(
                                             new AndCriteria(
-                                                new MetricCriteria("ATFD", OperationEnum.GREATER_THAN, 2),
-                                                new MetricCriteria("WMC", OperationEnum.GREATER_THAN, 47)),
-                                            new MetricCriteria("TCC", OperationEnum.LESS_THAN, 0.33)),
+                                                new MetricCriteria(CaDETMetric.ATFD , OperationEnum.GREATER_THAN, 2),
+                                                new MetricCriteria(CaDETMetric.WMC, OperationEnum.GREATER_THAN, 47)),
+                                            new MetricCriteria(CaDETMetric.TCC, OperationEnum.LESS_THAN, 0.33)),
                                   SmellType.GOD_CLASS);
             Rule rule2 = new Rule("10.1109/SCAM.2013.6648192",
                                   new OrCriteria(
-                                      new MetricCriteria("LOC", OperationEnum.GREATER_THAN, 750),
-                                      new MetricCriteria("[NOM] + [NOF]", OperationEnum.GREATER_THAN, 20)
+                                      new MetricCriteria(CaDETMetric.CLOC, OperationEnum.GREATER_THAN, 750),
+                                      new MetricCriteria(CaDETMetric.NMD_NAD, OperationEnum.GREATER_THAN, 20)
                                       ),
                                   SmellType.GOD_CLASS);
             Rule rule3 = new Rule("10.1109/MSR.2007.21",
                                   new OrCriteria(
-                                            new MetricCriteria("NMD", OperationEnum.GREATER_THAN, 15),
-                                            new MetricCriteria("NAD", OperationEnum.GREATER_THAN, 15)),
+                                            new MetricCriteria(CaDETMetric.NMD, OperationEnum.GREATER_THAN, 15),
+                                            new MetricCriteria(CaDETMetric.NAD, OperationEnum.GREATER_THAN, 15)),
                                   SmellType.GOD_CLASS);
             _rules = new List<Rule>();
             _rules.Add(rule1);
             _rules.Add(rule2);
             _rules.Add(rule3);
+            _dynamicRules = new List<Rule>();
+        }
+
+        private void createDynamicRules(List<CaDETClassDTO> caDetClassDtoList)
```

Comment:
createDynamicRules should start with a capital C (C# convention is PascalCase).

Furthermore, is "createDynamicRules" a good name, or is this a method that is more specialized - e.g., "DefineTopXMetricRules" (or a better name, this is of the top of my head).